File: Orimath/IO/PluginSetting.cs
using Sssl;
using System;
using System.Collections.Generic;

namespace Orimath.IO
{
    public sealed class PluginSetting
    {
        public string[] PluginOrder { get; set; } = Array.Empty<string>();

        public Dictionary<string, SsslObject> Settings { get; set; } = new Dictionary<string, SsslObject>();
    }
}

File: Orimath/IO/PluginSetting.fs
namespace Orimath.IO
open System
open System.Collections.Generic
open SsslFSharp

[<ReferenceEquality; NoComparison>]
type PluginSetting =
    {
        mutable PluginOrder: string[]
        mutable Settings: Dictionary<string, Sssl>
    }
    with

    static member CreateDefault() = { PluginOrder = array.Empty(); Settings = Dictionary() }

    member this.Clone() = this.MemberwiseClone() :?> PluginSetting

    interface ICloneable with
        member this.Clone() = this.MemberwiseClone()
